de_project/de_app/models.py

```diff
@@ -1,3 +1,206 @@
 from django.db import models
+from django.contrib.postgres.fields import ArrayField
+from datetime import datetime, timedelta
+from rest_framework import serializers
+from rest_framework.renderers import JSONRenderer
 
-# Create your models here.
+class Metric(models.Model):
```

Comment:
It's kinda weird that you have models which instantiate other models. I think it would make more sense if some of that logic was moved to helper classes.

Reply:
This cuts down on code duplication but we can consider alternatives in a future release.

Reply:
Moving some logic to a different class may slightly increase the number of lines, but it doesn't duplicate code.